### PR TITLE
Added 8 files for week2-js2 homework for review

### PR DESCRIPTION
**Pull request by TITHI1244**

### CODE REVIEW

File: javascript/javascript2/week2/main.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+"use strict"
+console.log("Script loaded");
+
+let products = getAvailableProducts();
+const lovedProducts = document.getElementById("loved-prducts");
+let lovedProductsNo = parseInt(lovedProducts.innerHTML);
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:6]:
```suggestion
let lovedProductsNumber = parseInt(lovedProducts.innerHTML);
```

File: javascript/javascript2/week2/main.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+// after clicking the 'like_product' button, remove it
+function addProduct(event) {
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:8]:
```suggestion
function addProduct(product) {
```
This function should probably just take a product and then add that product. Then the function can easier be reused

File: javascript/javascript2/week2/index.html [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      <button>Filter products</button>
+      <ul class="filter-contents">
+        <li><input type="number" id="search_min_price" name="search_min_price" placeholder="Set minimum price"></li>
+        <li><input type="number" id="search_max_price" name="search_max_price" placeholder="Set maximum price"></li>
+        <li><input type="number" id="search_products_rating" name="search_products_rating" placeholder="Set minimum rating"></li>
+        <li><input type="button" id="filter_button" name="filter_button" value="Filter"></li>
+      </ul>
+    </div>
+    <main id="products">
+      <div id="rendered_products">
```
Comment (benna100) [inline javascript/javascript2/week2/index.html:37]:
```suggestion
      <div id="rendered-products">
```

File: javascript/javascript2/week2/index.html [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+          <div id="icons">
+            <li><a href="#"><i class="fas fa-user"></i></a></li>
+            <li><a href="#"><i class="fas fa-heart"></i></a><span id="loved-prducts">0</span></li>
+          </div>         
+      </ul>
+    </nav>
+    <div class="filter">
+      <button>Filter products</button>
+      <ul class="filter-contents">
+        <li><input type="number" id="search_min_price" name="search_min_price" placeholder="Set minimum price"></li>
```
Comment (benna100) [inline javascript/javascript2/week2/index.html:30]:
```suggestion
        <li><input type="number" id="search-min-price" name="search-min-price" placeholder="Set minimum price"></li>
```

File: javascript/javascript2/week2/main.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+const inputElement3 = document.getElementById("search_max_price");
+const inputElement4 = document.getElementById("search_products_rating");
+const filterButton = document.getElementById("filter_button");
+// Set some initial values; for prices, if user does not specify any values, take the lowest and highest prices as min and maxPrice
+const productsPrice = products.map((product) => product.price);
+let minPrice = productsPrice.reduce((accumulator, currentValue) => (accumulator < currentValue) ? accumulator : currentValue, products[0].price);
+let maxPrice = productsPrice.reduce((accumulator, currentValue) => (accumulator > currentValue) ? accumulator : currentValue, products[0].price);
+let minRating = 0;
+
+function onNameChange(event) {
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:73]: SUper nice!

File: javascript/javascript2/week2/main.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    renderProducts(products.filter((product) => product.name.toLowerCase().startsWith(event.target.value.toLowerCase())));
+}
+// After clicking on filter button, set the predefined variebles as per user choice and call render method
+function onFilter() {
```
Comment (benna100) [inline javascript/javascript2/week2/main.js:77]: It's hard to figure out what this function should do from the name. Try give it a bit more descriptive name

**Review by benna100 — COMMENTED** [review]
Really good job! Try and focus on thinking about functions and what the responsibility of that function is. Fx what parameters makes sense that it takes. What should it do and what should it not do. 

Also give correct html class naming with dash - as seperator

File: javascript/javascript2/week2/main.js (hunk 1 above)
Comment (TITHI1244) [reply javascript/javascript2/week2/main.js:6]: Yes, got that.

File: javascript/javascript2/week2/index.html (hunk 3 above)
Comment (TITHI1244) [reply javascript/javascript2/week2/index.html:37]: Ok.

File: javascript/javascript2/week2/index.html (hunk 4 above)
Comment (TITHI1244) [reply javascript/javascript2/week2/index.html:30]: Got that.

File: javascript/javascript2/week2/main.js (hunk 6 above)
Comment (TITHI1244) [reply javascript/javascript2/week2/main.js:77]: Yes, will change that. Thank you for the feedback.

**Pull request merged**